app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt

```diff
@@ -332,9 +332,7 @@ class HomeFragment : Fragment(), AccountObserver {
     private fun handleTabAction(action: TabAction) {
         Do exhaustive when (action) {
             is TabAction.SaveTabGroup -> {
-                if ((activity as HomeActivity).browsingModeManager.isPrivate) {
-                    return
-                }
+                if ((activity as HomeActivity).browsingModeManager.mode.isPrivate) return
```

Comment:
Do we still have to grab it off the activity here?

Reply:
Perhaps it makes more sense to move this to a viewmodel instead of passing around the activity. What do you think?

Reply:
I'm going to leave the viewmodel change for a different PR.

Reply:
Sounds good to me 👍